tests/functionaltests/Tests/CortanaTests.mm

```diff
@@ -106,9 +222,85 @@
+    method_exchangeImplementations(originalMethod, swizzledMethod);
+
+    // Pass activation argument to method which activates the app
+    auto args = fakeVoiceCommandActivatedEventArgs.Detach();
+    UIApplicationActivationTest(reinterpret_cast<IInspectable*>(args));
+
+    // Un-Swizzle the methods now that we are done
+    method_exchangeImplementations(originalMethod, swizzledMethod);
+}
+
+TEST(CortanaTest, VoiceCommandForegroundActivationDelegateMethodsCalled) {
+    ASSERT_TRUE(willFinishLaunchingWithOptionsCalled);
+    ASSERT_TRUE(didFinishLaunchingWithOptionsCalled);
+    ASSERT_TRUE(didReceiveVoiceCommandCalled);
+
+    // Set flags to false so they can be used for other tests.
+    willFinishLaunchingWithOptionsCalled = false;
+    didFinishLaunchingWithOptionsCalled = false;
+    didReceiveVoiceCommandCalled = false;
+}
+
+// Creates test method which we call in TEST_CLASS_SETUP to activate app
+TEST(CortanaTest, ProtocolForegroundActivation) {
+    LOG_INFO("CortanaTest Protocol Foreground Activation Test: ");
+
+    // Create mocked data to pass into application
+    auto fakeUri = Make<MockUri>();
+    fakeUri->SetToString([](HSTRING* text) {
+        Wrappers::HString value;
+        value.Set(L"CORTANA_TEST");
+        *text = value.Detach();
+        return S_OK;
+    });
+
+    auto fakeProtocolActivatedEventArgs = Make<MockProtocolActivatedEventArgs>();
+    fakeProtocolActivatedEventArgs->Setget_Uri([&fakeUri](IUriRuntimeClass** uri) {
+        fakeUri.CopyTo(uri);
+        return S_OK;
+    });
+
+    fakeProtocolActivatedEventArgs->Setget_Kind([](ActivationKind* kind) {
+        *kind = ActivationKind_Protocol;
+        return S_OK;
+    });
+
+    fakeProtocolActivatedEventArgs->Setget_PreviousExecutionState([](ApplicationExecutionState* state) {
+        *state = ApplicationExecutionState_NotRunning;
+        return S_OK;
+    });
+
+    // Swizzle method to use our delegate for testing
+    Method originalMethod = class_getInstanceMethod([UIApplication class], @selector(setDelegate:));
+    Method swizzledMethod = class_getInstanceMethod([UIApplication class], @selector(swizzleProtocolDelegate:));
+    method_exchangeImplementations(originalMethod, swizzledMethod);
+
+    // Pass activation argument to method which activates the app
+    auto args = fakeProtocolActivatedEventArgs.Detach();
+    UIApplicationActivationTest(reinterpret_cast<IInspectable*>(args));
+
+    // Un-Swizzle the methods now that we are done
+    method_exchangeImplementations(originalMethod, swizzledMethod);
+}
+
+TEST(CortanaTest, ProtocolForegroundActivationDelegateMethodsCalled) {
+    ASSERT_TRUE(willFinishLaunchingWithOptionsCalled);
+    ASSERT_TRUE(didFinishLaunchingWithOptionsCalled);
+    ASSERT_TRUE(didReceiveProtocolCalled);
+
+    // Set flags to false so they can be used for other tests.
+    willFinishLaunchingWithOptionsCalled = false;
+    didFinishLaunchingWithOptionsCalled = false;
+    didReceiveProtocolCalled = false;
```

Comment:
This is a reason that we shouldn't be using static flags for test markers. Every test should be independent and static test flags like this make that easier to mess up than some scoped variables. As you can see from the MOCK_METHOD implementations, there are mechanism to dynamically control what something does at runtime. I would prefer if the CortanaProtocolForegroundTestDelegate class had similar functionality.